app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,43 @@
+class QuestionsController < ApplicationController
+    before_action :authenticate_user!, only: [:new, :create,:idnex,:show]
+
+    def show
+        @question = Question.find(params[:id])
+        @answer = Answer.new
+    end
+
+    def index
+        @questions = Questions.all.order(id: :asc)
+        @question = Question.new
+    end
+
+    def create
+        @question = Question.new(question_params)
+        if @question.save
+            flash[:notice] = "質問を投稿しました。"
+            redirect_to("/questions/index")
+        else
+            flash.now[:alert] = "エラーがあります！"
+            render("questions/index")
+    end
+
+    def edit 
+        @question=Question.find(params[:id])
+    end
+
+    def update
+        @question = Question.find(params[:id])
+        if @question.update(question_params)
+            flash[:notice] = "質問を編集しました"
+            redirect_to("/questions/#{@question.id}")
+        else
+            flash.now[:alert]="エラーがあります！"
+            render("questions/edit")
```

Comment:
`edit` に飛ばす場合は`id`が必要です。
```suggestion
            render("questions/#{@question.id}/edit")
```